crates/ruff/src/test.rs

```diff
@@ -219,7 +267,35 @@ fn print_diagnostics(diagnostics: Vec<Diagnostic>, file_path: &Path, source: &st
         })
         .collect();
 
-    print_messages(&messages)
+    if source_kind.is_jupyter() {
+        print_jupyter_messages(&messages, &filename, source_kind)
+    } else {
+        print_messages(&messages)
+    }
+}
+
+pub(crate) fn print_jupyter_messages(
```

Comment:
This seems a bit hacky to print cell information in the snapshot but just thought to put it out there to gain some feedback.

We could just avoid creating snapshots and test out using `assert_eq!` which is already being done for `cell_offsets`, `index` and `content`.

For reference, this is the commit which is isolated to this change (https://github.com/astral-sh/ruff/pull/5076/commits/f63103b20cd2fd6a0ad9ebbc2fa75fc0eb79f6f2)

\cc @charliermarsh @konstin

Reply:
i like it!

Reply:
I trust @konstin on this one, not gonna review closely ;)